### PR TITLE
Admin command for Network Blocklist

### PR DESCRIPTION
**Pull request by jordanschalm**
This PR enables modifying the network blocklist (added in https://github.com/onflow/flow-go/pull/3344) via admin command.
* Adds support for `[]Identifier`-typed configs to `ConfigManager`
* Registers existing setter/getter for blocklist to `ConfigManager`
* Updates documentation and corrects error return type (to `ValidationError`) in `Register*Config` setter field functions
* Moves instantiation of `ConfigManager` to instantiation of the node builder, so it is always accessible (same change made on `v0.28` in https://github.com/onflow/flow-go/pull/3555/commits/9926842eadfeb5c5148009e200461bf0b019eb7f)

#### Example Usage
Screen Shot 2022-11-11 at 9 36 28 AM
1. Add SN1 to blocklist of SN2 & restart SN2 to force re-connection
```sh
 % curl localhost:3704/admin/run_command -H 'Content-Type: application/json' -d '{"commandName": "set-config", "data": {"network-id-provider-blocklist": ["dcabeaa6709abafde476c7000a0261fa353151f560dd85e4ea56906bdd148050","cdba1cf3bc4e31f277bc7f02ad98f6ac2e036c69a1f4fe445e793daba646d276"]}}'

{"output":{"newValue":["dcabeaa6709abafde476c7000a0261fa353151f560dd85e4ea56906bdd148050","cdba1cf3bc4e31f277bc7f02ad98f6ac2e036c69a1f4fe445e793daba646d276"],"oldValue":[]}}
```
2. Clear SN2's blocklist
```sh
% curl localhost:3704/admin/run_command -H 'Content-Type: application/json' -d '{"commandName": "set-config", "data": {"network-id-provider-blocklist": []}}'

{"output":{"newValue":[],"oldValue":["cdba1cf3bc4e31f277bc7f02ad98f6ac2e036c69a1f4fe445e793daba646d276","dcabeaa6709abafde476c7000a0261fa353151f560dd85e4ea56906bdd148050"]}}
```
3. Restart SN1 to re-sync hotstuff views

### CODE REVIEW

**Review by AlexHentschel — APPROVED** [review]

File: module/updatable_configs/manager.go [hunk 1]
```diff
@@ -203,15 +220,56 @@ func (m *Manager) RegisterDurationConfig(name string, get GetDurationConfigFunc,
...
+	field := Field{
+		Name:     name,
+		TypeName: "IdentifierList",
+		Get: func() any {
+			return util.DetypeSlice(get().Strings())
+		},
+		Set: func(val any) error {
+			gval, ok := val.([]any)
+			if !ok {
+				return NewValidationErrorf("invalid type for duration config: %T", val)
```
Comment (peterargue) [inline module/updatable_configs/manager.go:256]:
```suggestion
				return NewValidationErrorf("invalid type for IdentifierList config: %T", val)
```

File: module/updatable_configs/manager_test.go [hunk 2]
```diff
@@ -101,4 +111,37 @@ func TestManager_RegisterDurationConfig(t *testing.T) {
...
+	// should be able to get the field
+	field, ok := mgr.GetField("field")
+	assert.True(t, ok)
+	// field must be parseable by structpb (otherwise admin server will error)
+	_, err = structpb.NewValue(field.Get())
+	require.NoError(t, err)
+
+	// should fail to set incorrect type
+	err = field.Set(struct{}{})
+	assert.Error(t, err)
```
Comment (peterargue) [inline module/updatable_configs/manager_test.go:136]: can you assert this is specifically a validation error. similar comment for other error cases in this test

**Review by peterargue — APPROVED** [review]
Looks goods other than a couple small comments.

**Comment by jordanschalm** [comment]
bors merge

**Pull request merged**